File: src/main/java/no/nav/familie/ks/sak/app/rest/Behandling/grunnlag/personinformasjon/RestPersonrelasjon.java
package no.nav.familie.ks.sak.app.rest.Behandling.grunnlag.personinformasjon;

import no.nav.familie.ks.sak.app.behandling.domene.grunnlag.personopplysning.PersonRelasjon;
import no.nav.familie.ks.sak.app.behandling.domene.kodeverk.RelasjonsRolleType;
import no.nav.familie.ks.sak.app.behandling.domene.typer.AktørId;

public class RestPersonrelasjon {
    public AktørId fraAktørId;
    public AktørId tilAktørId;
    public RelasjonsRolleType relasjonsrolle;
    public Boolean harSammeBosted;

    public RestPersonrelasjon(PersonRelasjon personRelasjon) {
        this.fraAktørId = personRelasjon.getFraAktørId();
        this.tilAktørId = personRelasjon.getTilAktørId();
        this.relasjonsrolle = personRelasjon.getRelasjonsrolle();
        this.harSammeBosted = personRelasjon.getHarSammeBosted();
    }
}

File: src/main/java/no/nav/familie/ks/sak/app/rest/behandling/grunnlag/personinformasjon/RestPersonrelasjon.kt
package no.nav.familie.ks.sak.app.rest.behandling.grunnlag.personinformasjon

import no.nav.familie.ks.sak.app.behandling.domene.grunnlag.personopplysning.PersonRelasjon
import no.nav.familie.ks.sak.app.behandling.domene.kodeverk.RelasjonsRolleType

data class RestPersonrelasjon(
        var fraPersonIdent: String,
        var tilPersonIdent: String,
        var relasjonsrolle: RelasjonsRolleType,
        var harSammeBosted: Boolean)

fun PersonRelasjon.toRestPersonrelasjon() = RestPersonrelasjon(
    fraPersonIdent = this.fraPersonIdent.ident,
    tilPersonIdent = this.tilPersonIdent.ident,
    relasjonsrolle = this.relasjonsrolle,
    harSammeBosted = this.harSammeBosted
)
